demos/python_demos/rag_chatbot/servable_stream/model.py

```diff
@@ -204,49 +185,140 @@ def serialize_completions(batch_size, result):
     return [Tensor("completion", serialize_byte_tensor(
         np.array(result, dtype=np.object_)).item())]
 
-
+def download_documents(file, target_folder):
+    if not os.path.exists(target_folder):
+        os.makedirs(target_folder)
+    print("file", file, "target_folder", target_folder)
+    file = open(file, "r")
+    for url in file.readlines():
+        url = url.strip()
+        if url.find('/'):
+            filename = url.rsplit('/', 1)[1]
+        else:
+            filename = url
+        print("downloading url", url, "filename", filename, "to", os.path.join(target_folder, filename),"#")
+        r = requests.get(url, allow_redirects=True, verify=False, proxies={"https": "http://proxy-igk.intel.com:911"})
+        if r.status_code == 200:
+            open(os.path.join(target_folder, filename), 'wb').write(r.content)
+            print("saved", filename)
+        else:
+            print("failed to download", url, "status code", r.status_code)
+
+def clean_target_folder(target_folder):
+    for file_path in os.listdir(target_folder):
+        print("removing", file_path)
+        os.remove(os.path.join(target_folder,file_path))
 
 class OvmsPythonModel:
     def initialize(self, kwargs: dict):
-        print(f"Loading LLM model {SELECTED_MODEL}...", flush=True)
+
+        device = os.environ.get("DEVICE", "CPU")   
+        llm_model_dir = os.environ.get("LLM_MODEL_DIR", SELECTED_MODEL)
+        llm_model_dir = os.path.join(kwargs["base_path"], llm_model_dir)  # use absolute path
+        print("llm model dir", llm_model_dir, flush=True)
+        model_name = llm_model_configuration["model_id"]
+        if os.path.isdir(llm_model_dir):
+            llm_model_2_load = llm_model_dir
+            export = False
+        else:
+            llm_model_2_load = model_name
+            export = True
+
+        print("llm model to load", llm_model_2_load, flush=True)
+        self.stop_tokens = llm_model_configuration.get("stop_tokens")
+        self.db = None
+        class_key = SELECTED_MODEL.split("-")[0]
+        self.tok = AutoTokenizer.from_pretrained(llm_model_2_load, trust_remote_code=True)
+
+        if self.stop_tokens is not None:
+            if isinstance(self.stop_tokens[0], str):
+                self.stop_tokens = self.tok.convert_tokens_to_ids(self.stop_tokens)
+            self.stop_tokens = [StopOnTokens(self.stop_tokens)]
+
+        from ov_llm_model import model_classes
```

Comment:
Do we still need this modeling class for the 3 models we document? (tiny, notus, etc)